lib/plugin/compilers/i18n.json.coffee

```diff
@@ -4,124 +4,147 @@ helpers = share.helpers
 compilers = share.compilers
 compiler_configuration = share.compiler_configuration
 
-compilers.i18n_json = (compileStep) ->
-  compiler_configuration.registerInputFile(compileStep)
```

Comment:
to my memory, you get the whole list and you are supposed to check their hashes to figure out what changed.

The CachingCompiler abstracts it away by letting you implement `compileSingleFile` method and then it would rely on hashes to cache. If you are not using CachingCompiler, then yeah, you would need to iterate over the list of matching files.